### ISSUE TITLE
Lag stays 1 using ConsumerGroupStream

### ISSUE THREAD

**Issue by stijnbe** [SOHU-Co/kafka-node#814]
When we are using ConsumerGroupStream and committing after each message is processed, the lag per partition stays at 1 until we close the consumer. As most of our monitoring is based on this lag I was wondering if this is normal behavior. The issue is also there when using autocommit.

### Environment
* Node version: 8.9.1
* Kafka-node version: master #7426cf9
* Kafka version: 0.10.2.1

### Code used
```
const {ConsumerGroupStream} = require('kafka-node');
const logger = require('./../logger');
const {Writable} = require("stream");

const connect = (connectionString, groupId, topics) => {
    const consumerGroupOptions = {
        host: connectionString,
        groupId: groupId,
        sessionTimeout: 15000,
        protocol: ['roundrobin'],
        autoCommit: true
    };
    logger.info(`Listening for the ${topics} messages...`);

    return new ConsumerGroupStream(consumerGroupOptions, topics);
};

const consume = (connectionString, groupId, topics, onMessage) => {
    const consumer = connect(connectionString, groupId, topics);

    const processor = new Writable({
        objectMode: true,
        write(message, encoding, done) {
            const payload = JSON.parse(message.value);
            onMessage(payload, function(){
                consumer.commit(message, true, function(){
                  done();
                });
            });
        }
    });

    consumer.pipe(processor);

    return consumer;
};

module.exports = {
    consume: consume
};
```

**Comment by skoval-dev** [comment]
Hi @stijnbe, 
Also I've noticed that when I use ConsumerGroupStream - consumer GroupID doesn't appear in kafka lag and in kafka-consumer-groups list. 
Just checked it with HighLevelConsumer and see that it works okay. 
Don't you have the same problem with consumer detection in kafka lag ?

**Comment by stijnbe** [comment]
@skoval-dev We don't have any issues with that. Monitoring the lag works fine.

**Comment by sanditha** [comment]
Hi, even i am facing same issue. 
while using consumerGroupStream, lag always remains one . when i restart. the last message comes again.
Does not commit the last message. Please help me with this

**Comment by hyperlink** [comment]
I will look into this bug thanks for reporting it.

**Comment by sanditha** [comment]
This is my code:
 this.consumerStreamGroup.pipe(
      through2.obj(function (data: any, enc: any, done: any) {
        console.log("data", data);
        self.consumerStreamGroup.commit(data, true, function (err: Error, message: any) {
          console.log("done************");
          done();
        });
}));
console.log("message buffer length after commit:::", self.consumerStreamGroup.messageBuffer.length)

Logs:
  kafka-node:ConsumerGroupStream committing [ { topic: 'LICENSE-TEST', partition: 0, offset: 6320, metadata: 'm' } ] +3m
this.messageBuffer after commit::: 0
  kafka-node:ConsumerGroupStream _read called +4ms
done************

- If u see logs, commit happens  and also callback is called but still lag is 1.

**Comment by sanditha** [comment]
Thanks @hyperlink . is it available in version 2.3.1?. I don't see the changes on npm install

**Comment by hyperlink** [comment]
@sanditha I just published it as 2.3.2.

**Comment by sanditha** [comment]
Awesome :)
